### PR TITLE
Recorded Future: Examples and bug fixes

### PR DESCRIPTION
**Pull request by jschipp-r7**
### Description

Describe the proposed changes:

  - Add example inputs
  - Add example outputs
  - Fix output bug in List actions (successfully tested)

### Testing

Any testing information goes here.

```
$ make validate
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Colors.mk ../tools/Makefiles/Helpers.mk
--
[*] Running validators
You are using pip version 18.1, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 546.928ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting
```

### CODE REVIEW

File: recorded_future/help.md [hunk 1]
```diff
@@ -26,9 +26,19 @@ returned in the [STIX](https://stixproject.github.io/about/) format.
...
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|api_key|credential_secret_key|None|True|API Key|None|9de5069c5afe602b2ea0a04b66beb2c0|
+
+Example input:
+
+```
+{
+  "api_key": {
+      "secretKey": "9de5069c5afe602b2ea0a04b66beb2c0"
```
Comment (mrinehart-r7) [inline recorded_future/help.md:38]:
```suggestion
      "secretKey": "9de5069c5afe602b2ea0a04b66beb2c0"
```
This is sanitized right?

Comment (jschipp-r7) [reply recorded_future/help.md:38]:
```
$ echo "Rapid7 InsightConnect" | md5
9de5069c5afe602b2ea0a04b66beb2c0
```

File: recorded_future/help.md [hunk 2]
```diff
@@ -80,99 +172,330 @@ _This action does not contain any inputs._
...
+
+Example input:
+
+```
+{
+  "direction": "asc",
+  "freetext": "example",
+  "from": 1,
+  "limit": 10,
+  "orderby": "Created"
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:234]:
```suggestion
  "orderby": Riskscore"
```

File: recorded_future/help.md [hunk 3]
```diff
@@ -80,99 +172,330 @@ _This action does not contain any inputs._
...
+
+Example input:
+
+```
+{
+  "direction": "asc",
+  "freetext": "CVE",
+  "from": 1,
+  "limit": 10,
+  "orderby": "Created"
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:382]:
```suggestion
  "orderby": "Riskscore"
```

File: recorded_future/help.md [hunk 4]
```diff
@@ -80,99 +172,330 @@ _This action does not contain any inputs._
...
+|ip_range|string|None|True|IP address range to search|None|198.51.100.0/24|
+|limit|number|None|True|Number of results to retrieve, up to 100|None|10|
+|orderby|string|None|True|Which property to sort the results by|['Created', 'Lastseen', 'Firstseen', 'Modified', 'Riskscore', 'Rules', 'Sevendayshits', 'Sixtydayshits', 'Totalhits']|Lastseen|
+
+Example input:
+
+```
+{
+  "direction": "asc",
+  "from": 2,
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:450]:
```suggestion
  "from": 0,
```

File: recorded_future/help.md [hunk 5]
```diff
@@ -186,88 +509,206 @@ _This action does not contain any inputs._
...
-|list|string|None|False|The risk list to retrieve|['large', 'linked_to_attack_vector', 'linked_to_cyber_attack', 'linked_to_malware', 'linked_to_vulnerability', 'positive_malware_verdict', 'threat_researcher']|
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|list|string|None|False|The risk list to retrieve|['large', 'linked_to_attack_vector', 'linked_to_cyber_attack', 'linked_to_malware', 'linked_to_vulnerability', 'positive_malware_verdict', 'threat_researcher']|positive_malware_verdict|
+
+Example input:
+
+```
+{
+  "list": "linked_to_attack_vector"
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:553]:
```suggestion
  "list": "positive_malware_verdict"
```

File: recorded_future/help.md [hunk 6]
```diff
@@ -186,88 +509,206 @@ _This action does not contain any inputs._
...
+|direction|string|asc|True|Sort results ascending/descending|['asc', 'desc']|asc|
+|from|number|0|True|Number of initial records to skip|None|0|
+|limit|number|None|True|Number of results to retrieve, up to 100|None|10|
+|orderby|string|None|True|Which property to sort the results by|['Created', 'Lastseen', 'Firstseen', 'Lastseen', 'Modified', 'Riskscore', 'Rules', 'Sevendayshits', 'Sixtydayshits', 'Totalhits']|Riskscore|
+
+Example input:
+
+```
+{
+  "algorithm": "MD5",
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:581]:
```suggestion
  "algorithm": "ShA-1",
```

File: recorded_future/help.md [hunk 7]
```diff
@@ -186,88 +509,206 @@ _This action does not contain any inputs._
...
+  "direction": "asc",
+  "from": 2,
+  "limit": 10,
+  "orderby": "Created"
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:585]:
```suggestion
  "orderby": "Riskscore"
```

File: recorded_future/help.md [hunk 8]
```diff
@@ -308,11 +749,21 @@ This action is used to retrieve information about a specified hash.
...
+|comment|string|None|False|Add a comment to a hash|None|Hash look up performed by InsightConnect|
+|fields|[]string|None|False|List of fields to include with results e.g ["timestamps", "sightings", "threatLists", "analystNotes", "counts", "entity", "hashAlgorithm", "intelCard", "metrics", "relatedEntities", "risk", "timestamps"]|None|['risk', 'timestamps', 'sightings']|
+|hash|string|None|True|Hash|None|44d88612fea8a8f36de82e1278abb02f|
+
+Example input:
+
+```
+{
+  "comment": "Hash look up performed by InsightConnect",
+  "hash": "478c076749bef74eaf9bed4af917aee228620b23",
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:763]:
```suggestion
  "hash": "44d88612fea8a8f36de82e1278abb02f",
```

File: recorded_future/help.md [hunk 9]
```diff
@@ -380,11 +831,21 @@ This action is used to retrieve information about a specified URL.
...
+|comment|string|None|False|Add a comment to an IP address lookup for Recorded Future|None|URL look up performed by InsightConnect|
+|fields|[]string|None|False|List of fields to include with results e.g ["sightings", "analystNotes", "counts", "entity", "metrics", "relatedEntities", "risk", "timestamps"]|None|['relatedEntities', 'risk', 'sightings']|
+|url|string|None|True|URL|None|https://example.com|
+
+Example input:
+
+```
+{
+  "comment": "URL look up performed by InsightConnect",
+  "url": "http://www.example.com",
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:845]:
```suggestion
  "url": "https://www.example.com",
```

File: recorded_future/help.md [hunk 10]
```diff
@@ -422,27 +883,215 @@ This action is used to fetch a risk list of the IP addresses that match a specif
...
+
+Example input:
+
+```
+{
+  "list": "current_cc_server"
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:894]:
```suggestion
  "list": "malware_delivery"
```

File: recorded_future/help.md [hunk 11]
```diff
@@ -422,27 +883,215 @@ This action is used to fetch a risk list of the IP addresses that match a specif
...
+|IP_address|string|None|True|IP address|None|198.51.100.1|
+|comment|string|None|False|Add comment to IP address lookup for Recorded Future|None|IP look up performed by InsightConnect|
+|fields|[]string|None|False|List of fields to include with results e.g ["sightings", "threatLists", "analystNotes", "counts", "entity", "intelCard", "metrics", "relatedEntities", "riskyCIDRIPs","risk", "location", "timestamps"]|None|['riskyCIDRIPs', 'risk', 'sightings']|
+
+Example input:
+
+```
+{
+  "comment": "IP look up performed by InsightConnect",
+  "IP_address": "198.51.100.100",
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:1091]:
```suggestion
  "IP_address": "198.51.100.1",
```

File: recorded_future/help.md [hunk 12]
```diff
@@ -484,15 +1133,76 @@ This action is used to fetch a risk list of vulnerabilities matching a specified
...
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|list|string|None|False|The risk list to retrieve|['cyber_exploit_signal_critical', 'cyber_exploit_signal_important', 'cyber_exploit_signal_medium', 'historical_scanner_uptake', 'historically_linked_to_exploit_kit', 'historically_linked_to_malware', 'historically_linked_to_ransomware', 'historically_linked_to_remote_access_trojan', 'large', 'linked_to_recent_cyber_exploit', 'nist_severity_critical', 'nist_severity_high', 'nist_severity_low', 'nist_severity_medium', 'recent_scanner_uptake', 'recently_linked_to_exploit_kit', 'recently_linked_to_malware', 'recently_linked_to_ransomware', 'recently_linked_to_remote_access_trojan', 'web_reporting_prior_to_nvd_disclosure']|nist_severity_critical|
+
+Example input:
+
+```
+{
+  "list": "cyber_exploit_signal_critical"
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:1144]:
```suggestion
  "list": "nist_severity_critical"
```

File: recorded_future/help.md [hunk 13]
```diff
@@ -506,27 +1216,81 @@ _This action does not contain any inputs._
...
-|id|string|None|True|CVE or RF ID|None|
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|id|string|None|True|CVE or RF ID|None|CVE-2014-0160|
+
+Example input:
+
+```
+{
+  "id": "CVE-2017-0147"
```
Comment (tylerslijboom-r7) [inline recorded_future/help.md:1262]:
```suggestion
  "id": "CVE-2014-0160"
```

File: recorded_future/help.md (hunk 8 above)
Comment (jschipp-r7) [reply recorded_future/help.md]: Using EICAR test MD5 hash here `44d88612fea8a8f36de82e1278abb02f` , updated

File: recorded_future/help.md (hunk 11 above)
Comment (jschipp-r7) [reply recorded_future/help.md:1091]: Applied manually

**Pull request merged**